proto/spacemesh/smesher_types.proto

```diff
@@ -1,55 +1,52 @@
 syntax = "proto3";
 package spacemesh;
 
-// Data on commitment file
-message CommitmentFile {
-    string path = 1;                // Path to post file(s) on a volume
-    uint64 commitment_size = 2;     // Size of commitment data (in bytes)
-}
 
-// User-provided post data
-message CommitmentUserData {
-    CommitmentFile file_data = 1;   // Data on commitment file
-    uint32 processor_id = 2;        // Processor to be used for smeshing, e.g., cpu, gpu0, gpu1, etc.
-    bool   append = 3;              // Append to existing file if it exists. Otherwise overwrite.
-    bool   throttle = 4;            // Throttle down init phase computations while user is interactive on system
+// Bit flags for compute engine selection
+enum ComputeEngineFlag {
+    NONE = 0;
+    CPU = 1; // useful for testing on systems without a cuda or vulkan GPU
+    CUDA = 2;
+    VULKAN = 4;
 }
 
-// A commitment init processor, e.g., a cpu or a gpu
-message CommitmentProcessor {
-    uint32 id = 1;                  // Numerical identifier - unique per system
-    string processor_name = 2;      // Proc info, e.g., Nvidia `Geforce RTX`, `AMD Radeon`, `Intel Genuine i7 CPU`
-    string api = 3;                 // Processing api to be used with the processor, e.g., `Vulkan`, `Cuda` or `x86-64`
+message ComputeEngineFlags {
+    uint32 flags = 1;
 }
 
-// A list of CommitmentProcessor
-message CommitmentProcessors {
-    repeated CommitmentProcessor processors = 1;
+// Basic post data. Used for users to provide requested post data and by
+// other messages which contain post data.
+message PostData {
+    string path = 1; // User provided path to create the post data files at
+    uint64 data_size = 2; // Requested post data size
+    bool   append = 3; // Append to existing files if they exist. Otherwise overwrite.
+    bool   throttle = 4; // Throttle down setup phase computations while user is interactive on ///system
+    uint32 compute_engine_flags = 5; // One ore more available compute engines to be used for gpu post
```

Comment:
```suggestion
    ComputeEngineFlags flags = 5; // One ore more available compute engines to be used for gpu post
```

Reply:
Unfortunately, this won't work as the ComputeEngineFlags is just an enum and the uint32 is needed to represent a bit field of one or more of these flags set. I wish had support for this but it doesn't. Used best practices after reading on bit fields in protobufs...

Reply:
updated